tensorzero-core/src/endpoints/datasets/v1/types.rs

```diff
@@ -151,6 +151,55 @@ pub struct UpdateJsonDatapointRequest {
     pub metadata: DatapointMetadataUpdate,
 }
 
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, ts_rs::TS)]
+#[ts(export, optional_fields)]
+#[export_schema]
+#[serde(untagged)]
+pub enum InferenceExtraBodyProviderType {
+    #[schemars(title = "InferenceExtraBodyProviderTypeProvider")]
+    Provider { model_provider_name: String },
+
+    #[schemars(title = "InferenceExtraBodyProviderTypeVariant")]
+    Variant { variant_name: String },
+
+    #[schemars(title = "InferenceExtraBodyProviderTypeModelProvider")]
+    ModelProvider {
+        model_name: String,
+        provider_name: Option<String>,
+    },
+
+    #[schemars(title = "InferenceExtraBodyProviderTypeAlways")]
+    Always { always: bool },
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, JsonSchema, ts_rs::TS)]
+#[ts(export, optional_fields)]
+#[export_schema]
+#[serde(tag = "type", rename_all = "snake_case")]
+pub enum InferenceExtraBodyReplacementKind {
+    /// Replaces the field with the given value.
+    #[schemars(title = "InferenceExtraBodyReplacementKindValue")]
+    Value { value: Value },
+
+    /// Deletes the field.
+    #[schemars(title = "InferenceExtraBodyReplacementKindDelete")]
+    Delete { delete: bool },
```

Comment:
I'm starting to think that unit variants or enums with tuple values are a mistake for wire types - you can't express them in JSON.